packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -0,0 +1,938 @@
+   */
+  LicenseIncluded = 'license-included',
+
+  /**
+   * Bring your own licencse.
+   */
+  BringYourOwnLicense = 'bring-your-own-license',
+
+  /**
+   * General public license.
+   */
+  GeneralPublicLicense = 'general-public-license'
+}
+
+/**
+ * The processor features.
+ */
+export interface ProcessorFeatures {
+  /**
+   * The number of CPU core.
+   */
+  readonly coreCount?: number;
+
+  /**
+   * The number of threads per core.
+   */
+  readonly threadsPerCore?: number;
+}
+
+/**
+ * The type of storage.
+ */
+export enum StorageType {
+  /**
+   * Standard.
+   */
+  Standard = 'standard',
+
+  /**
+   * General purpose (SSD).
+   */
+  GP2 = 'gp2',
+
+  /**
+   * Provisioned IOPS (SSD).
+   */
+  IO1 = 'io1'
+}
+
+/**
+ * The retention period for Performance Insight.
+ */
+export enum PerformanceInsightRetentionPeriod {
+  /**
+   * Default retention period of 7 days.
+   */
+  Default = 7,
+
+  /**
+   * Long term retention period of 2 years.
+   */
+  LongTerm = 731
+}
+
+/**
+ * Construction properties for a DatabaseInstanceNew
+ */
+export interface DatabaseInstanceNewProps {
+  /**
+   * The name of the compute and memory capacity classes.
+   */
+  readonly instanceClass: ec2.InstanceType;
+
+  /**
+   * Specifies if the database instance is a multiple Availability Zone deployment.
+   *
+   * @default false
+   */
+  readonly multiAz?: boolean;
+
+  /**
+   * The name of the Availability Zone where the DB instance will be located.
+   *
+   * @default no preference
+   */
+  readonly availabilityZone?: string;
+
+  /**
+   * The storage type.
+   *
+   * @default GP2
+   */
+  readonly storageType?: StorageType;
+
+  /**
+   * The number of I/O operations per second (IOPS) that the database provisions.
+   * The value must be equal to or greater than 1000.
+   *
+   * @default no iops
+   */
+  readonly iops?: number;
+
+  /**
+   * The number of CPU cores and the number of threads per core.
+   *
+   * @default no processor features
+   */
+  readonly processorFeatures?: ProcessorFeatures;
+
+  /**
+   * A name for the DB instance. If you specify a name, AWS CloudFormation
+   * converts it to lowercase.
+   *
+   * @default a CloudFormation generated name
+   */
+  readonly instanceIdentifier?: string;
+
+  /**
+   * The VPC network where the DB subnet group should be created.
+   */
+  readonly vpc: ec2.IVpcNetwork;
+
+  /**
+   * The type of subnets to add to the created DB subnet group.
+   *
+   * @default private
+   */
+  readonly vpcPlacement?: ec2.SubnetSelection;
+
+  /**
+   * The port for the instance.
+   */
+  readonly port?: number;
+
+  /**
+   * The option group to associate with the instance.
+   *
+   * @default no option group
+   */
+  readonly optionGroup?: IOptionGroup;
+
+  /**
+   * Whether to enable mapping of AWS Identity and Access Management (IAM) accounts
+   * to database accounts.
+   *
+   * @default false
+   */
+  readonly enableIAMDatabaseAuthentication?: boolean;
```

Comment:
Would it be sufficient to just call this `iamAuth` or something more concise?

Reply:
I would go for `iamAuthentication`

Reply:
Ideally, there should also be a `grant` method to provide permissions there. And this is tricky because it requires to know the instance's *Resource ID*, which I don't reckon is surfaced by CloudFormation (aka requires CustomResource 😔)

Reply:
> aka requires CustomResource

Another use case for https://github.com/awslabs/aws-cdk/pull/1850